src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndex.java

```diff
@@ -38,7 +70,51 @@ public abstract void saveIndex(int[] ids, float[][] data, String indexPath, Stri
      *
      * @return size of the hnsw index on the disk in KB.
      */
-    public abstract long getIndexSize();
+    public long getIndexSize() {
+        return this.indexSize;
+    }
+
+    public SpaceType getSpaceType() {
+        return this.spaceType;
+    }
+
+    public void close() {
+        Lock writeLock = readWriteLock.writeLock();
+        writeLock.lock();
+        // Autocloseable documentation recommends making close idempotent. We don't expect to doubly close
+        // but this will help prevent a crash in that situation.
+        if (this.isClosed) {
+            return;
+        }
+        try {
+            gcJniWrapper(this.indexPointer);
+        } finally {
+            this.isClosed = true;
+            writeLock.unlock();
+        }
+    }
+
+    /**
+     * determines the size of the engine index on disk
+     * @param indexPath absolute path of the index
+     *
+     */
+    protected static long computeFileSize(String indexPath) {
+        if (indexPath == null || indexPath.isEmpty()) {
+            return 0;
+        }
+        File file = new File(indexPath);
+        if (!file.exists() || !file.isFile()) {
+            return 0;
+        }
 
-    public abstract void close();
+        return file.length() / 1024 + 1;
```

Comment:
let's create constant to understand what does 1024 is and why add 1? Also, method can be named as getSizeInKB(String filePath)

Reply:
Good point. Will update.